src/main/java/com/squareup/kotlinpoet/Util.kt

```diff
@@ -113,3 +113,58 @@ internal fun stringLiteralWithQuotes(value: String): String {
     return result.toString()
   }
 }
+
+internal fun isIdentifier(name: String): Boolean {
+  return IDENTIFIER_REGEX.matches(name)
+}
+
+internal fun isKeyword(name: String): Boolean {
+  return KEYWORDS.contains(name)
+}
+
+internal fun isName(name: String): Boolean {
+  return NAME_REGEX.split(name, 0).toTypedArray().none { isKeyword(name) }
+}
+
+private val NAME_REGEX = "\\.".toRegex()
+private val IDENTIFIER_REGEX
+    = ("((\\p{gc=Lu}+|\\p{gc=Ll}+|\\p{gc=Lt}+|\\p{gc=Lm}+|\\p{gc=Lo}+|\\p{gc=Nl}+)+" +
+    "\\d*" +
+    "\\p{gc=Lu}*\\p{gc=Ll}*\\p{gc=Lt}*\\p{gc=Lm}*\\p{gc=Lo}*\\p{gc=Nl}*)" +
```

Comment:
I'm not sure what to do here.
We could duplicate Character.isLetter as a helper and use a loop.
Any ideas?
I could also revert this and only replace `isKeyword` and `isName` until we think of something sane.